### PR TITLE
NestedFolders: New Browse Dashboards views

### PR DESCRIPTION
**Pull request by joshhunt**
**What is this feature?**

Scaffolds out pages to build the new Browse Dashboards for nested folders.

At the moment these are on a seperate `/nested-dashboards` route, but once mature we'll switch this over to the Browse views.

Part of #65604

### CODE REVIEW

File: public/app/routes/routes.tsx [hunk 1]
```diff
@@ -513,7 +517,7 @@ export function getAppRoutes(): RouteDescriptor[] {
     },
     // TODO[Router]
     // ...playlistRoutes,
-  ];
+  ].filter(isTruthy);
```
Comment (ashharrison90) [inline public/app/routes/routes.tsx:520]: minor question - is there an advantage to doing this over `.filter(Boolean)`? 🤔

File: public/app/features/browse-dashboards/components/SearchView.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { parseRouteParams } from 'app/features/search/utils';
+
+interface SearchViewProps {
+  searchState: ReturnType<typeof parseRouteParams>;
+}
+
+export default function SearchView({ searchState }: SearchViewProps) {
```
Comment (ashharrison90) [inline public/app/features/browse-dashboards/components/SearchView.tsx:9]: minor nit: think we [prefer named exports](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#exports)

File: public/app/features/browse-dashboards/components/BrowseView.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { getFolderChildren } from 'app/features/search/service/folders';
+import { DashboardViewItem } from 'app/features/search/types';
+
+type NestedData = Record<string, DashboardViewItem[] | undefined>;
+
+interface BrowseDashboardsViewProps {
+  folderUID: string | undefined;
+}
+
+export default function BrowseDashboardsView({ folderUID }: BrowseDashboardsViewProps) {
```
Comment (ashharrison90) [inline public/app/features/browse-dashboards/components/BrowseView.tsx:13]: minor nit: think we [prefer named exports](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#exports)

File: public/app/features/browse-dashboards/components/BrowseActions.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React, { useMemo } from 'react';
+
+import { Input } from '@grafana/ui';
+import { ActionRow } from 'app/features/search/page/components/ActionRow';
+import { SearchLayout } from 'app/features/search/types';
+
+export default function BrowseActions() {
```
Comment (ashharrison90) [inline public/app/features/browse-dashboards/components/BrowseActions.tsx:7]: minor nit: think we [prefer named exports](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#exports)

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        {searchState.query ? <SearchView searchState={searchState} /> : <BrowseView folderUID={folderUID} />}
+      </Page.Contents>
+    </Page>
+  );
+});
+
+BrowseDashboardsPage.displayName = 'BrowseDashboardsPage';
+
+export default BrowseDashboardsPage;
```
Comment (ashharrison90) [inline public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:49]: minor nit: think we [prefer named exports](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#exports)

**Review by ashharrison90 — APPROVED** [review]
some minor nits/questions, otherwise looks like a good start 👍

File: public/app/routes/routes.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/routes/routes.tsx:520]: [isTruthy](https://github.com/grafana/grafana/blob/eea35b7e2f5072bd50584080c7b15b3609772e48/packages/grafana-data/src/types/data.ts/#L196) is a type guard, so this satisfies Typescript to have no empty values in the array.

**Pull request merged**